app/workers/stale_issue_marker.rb

```diff
@@ -23,49 +23,49 @@ class StaleIssueMarker
 Thank you for all your contributions!
   EOS
 
-  attr_reader :fq_repo_name
-
-  def perform(fq_repo_name)
-    @fq_repo_name = fq_repo_name
-    raise "The label #{STALE_LABEL_NAME} does not exist on #{fq_repo_name}" unless GithubService.valid_label?(fq_repo_name, STALE_LABEL_NAME)
-
-    GithubService.issues(fq_repo_name, :state => :open, :sort => :updated, :direction => :asc).each do |issue|
-      pinned = (issue.labels & PINNED_LABELS).present?
-
-      if issue.updated_at < stale_date && !pinned
-        if issue.pull_request?
-          closable_prs << issue
-        else
-          stale_issues << issue
-        end
+  def perform
+    GithubService.search_issues(*search_args).each do |issue|
+      if issue.pull_request?
+        close_pr(issue)
+      else
+        mark_as_stale(issue)
       end
     end
-
-    stale_issues.each do |issue|
-      next if issue.labels.include?(STALE_LABEL_NAME)
-      logger.info("[#{Time.now.utc}] - Marking issue #{fq_repo_name}##{issue.number} as stale")
-      issue.add_labels([STALE_LABEL_NAME])
-      issue.add_comment(STALE_ISSUE_MESSAGE)
-    end
-
-    closable_prs.each do |pr|
-      logger.info("[#{Time.now.utc}] - Closing stale PR #{fq_repo_name}##{pr.number}")
-      GithubService.close_pull_request(pr.fq_repo_name, pr.number)
-      pr.add_comment(CLOSABLE_PR_MESSAGE)
-    end
   end
 
   private
 
+  def search_args
+    query  = "is:open archived:false update:<#{stale_date.strftime('%Y-%m-%d')}"
+    query << " #{PINNED_LABELS.map { |label| %Q{-label:"#{label}"} }.join(" ")}"
+    query << " #{enabled_repo_names.map { |repo| %Q{repo:"#{repo}"} }.join(" ")}"
+
+    [query, {:sort => :updated, :direction => :asc}]
+  end
+
   def stale_date
     @stale_date ||= 6.months.ago
   end
 
-  def stale_issues
-    @stale_issues ||= []
+  def repo_has_label(repo)
```

Comment:
For what it is worth, I am not 100% sold on this, but decided as a start to try and keep the previous functionality as best as possible.

In my opinion, I think we should probably automate our label creation for repos so a check like this isn't required.

Reply:
We do automate our label creation, but accidents happen...would prefer having the code be safe

Reply:
For a future update, I might consider doing this in one fell swoop in `perform` just so we aren't have to check against the same repo every time (though, labels are cached, so it is more a "avoid extra useless work" fix instead of a "avoid `N+1`s"), but I think it is simpler for this pass to just keep it as is instead of rocking the boat with an extra fix.

Will resolve for now.